server/broadcast.go

```diff
@@ -347,13 +359,30 @@ func transcodeSegment(cxn *rtmpConnection, seg *stream.HLSSegment, name string,
 	// send segment to the orchestrator
 	glog.V(common.DEBUG).Infof("Submitting segment nonce=%d manifestID=%s seqNo=%d orch=%s", nonce, cxn.mid, seg.SeqNo, sess.OrchestratorInfo.Transcoder)
 
-	res, err := SubmitSegment(sess, seg, nonce)
-	if err != nil || res == nil {
+	res, tErr := SubmitSegment(sess, seg, nonce)
+	// If TicketParams have expired, refresh them
+	if tErr != nil && tErr == pm.ErrTicketParamsExpired {
+		glog.V(common.VERBOSE).Infof("ticketparams expired, refreshing for orchestrator=%v", sess.OrchestratorInfo.Transcoder)
+		oURL, err := url.Parse(sess.OrchestratorInfo.Transcoder)
+		if err != nil {
+			cxn.sessManager.removeSession(sess)
+			return nil, fmt.Errorf("invalid orchestrator URL url=%v", sess.OrchestratorInfo.Transcoder)
+		}
+		oInfo, err := getOrchestratorInfoRPC(context.Background(), sess.Broadcaster, oURL)
```

Comment:
We need to pass in a context with deadline here or else this will never time out.

Reply:
What would be a sensible timeout value here? I currently set it at 2.5 sec in afb03034

Reply:
oh sorry , I just checked the code snipped below. Nvm